sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientCodeSnippets.java

```diff
@@ -0,0 +1,109 @@
+            .buildClient();
+
+        //create TableClient
+        TableClient tableClient = new TableClientBuilder()
+            .connectionString("connectionString")
+            .tableName("OfficeSupplies")
+            .buildClient();
+
+
+        //create a table
+        try {
+            tableServiceClient.createTable("OfficeSupplies");
+        } catch (TableServiceErrorException e) {
+            //use azure core errors? based on
+            System.out.println("Create Table Unsuccessful. Error: " + e);
+        }
+
+        //delete  table
+        try {
+            tableServiceClient.deleteTable("OfficeSupplies");
+        } catch (TableServiceErrorException e) {
+            System.out.println("Delete Table Unsuccessful. Error: " + e);
+        }
+
+        //query tables
+        String selectString = "$selectString= TableName eq 'OfficeSupplies'";
+
+        try {
+            List<AzureTable> responseTables = tableServiceClient.queryTables(selectString);
+        } catch (HttpResponseException e) {
+            System.out.println("Table Query Unsuccessful. Error: " + e);
+        }
+
+
+        //insert entity
+        String tableName = "OfficeSupplies";
+        String row = "crayola markers";
+        String partitionKey = "markers";
+        HashMap<String, Object> tableEntityProperties = new HashMap<>();
+        TableEntity tableEntity = new TableEntity(tableName, row, partitionKey, tableEntityProperties);
+        try {
+            tableEntity = tableClient.insertEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Insert Entity Unsuccessful. Error: " + e);
+        }
+
+
+        //update entity
+        tableEntity.addProperty("Seller", "Crayola");
+        try {
+            tableClient.updateEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Update Entity Unsuccessful. Error: " + e);
+        }
+
+
+        //upsert entity (where it is an insert or replace)
+        tableEntity.addProperty("Price", "5");
+        try {
+            tableClient.insertOrReplaceEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Upsert Entity Unsuccessful. Error: " + e);
+        }
+
+        //upsert entity (where it is an insert or merge)
+        tableEntity.addProperty("Price", "5");
+        try {
+            tableClient.insertOrMergeEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Upsert Entity Unsuccessful. Error: " + e);
+        }
+
+
+        //delete entity
+        try {
+            tableClient.deleteEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Delete Entity Unsuccessful. Error: " + e);
+        }
+
+
+        //query a table
+        String filterString2 = "$filter = Product eq 'markers'";
+        String selectString2 = "$select = Seller eq 'crayola'";
+        try {
```

Comment:
Should we also expose the `top` OData parameter? I'm of the mind we should either have 3 params, one for each OData parameter, or only a single query parameter (leaning towards the second option since we can make a QueryBuilder to help create query strings).